### PR TITLE
[SCB-905] Add global rest failure handler

### PR DESCRIPTION
**Pull request by yhs0092**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
See details in [SCB-905](https://issues.apache.org/jira/browse/SCB-905)

### CODE REVIEW

File: transports/transport-rest/transport-rest-vertx/src/main/java/org/apache/servicecomb/transport/rest/vertx/RestServerVerticle.java [hunk 1]
```diff
@@ -117,6 +123,23 @@ public void start(Future<Void> startFuture) throws Exception {
     }
   }
 
+  private void mountGlobalRestFailureHandler(Router mainRouter) {
+    GlobalRestFailureHandler globalRestFailureHandler =
+        SPIServiceUtils.getPriorityHighestService(GlobalRestFailureHandler.class);
+    Handler<RoutingContext> failureHandler = null == globalRestFailureHandler ?
+        ctx -> {
+          LOGGER.error("unexpected failure happened", ctx.failure());
+          ctx.response().setStatusCode(500).putHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN)
```
Comment (liubao68) [inline transports/transport-rest/transport-rest-vertx/src/main/java/org/apache/servicecomb/transport/rest/vertx/RestServerVerticle.java]: It's better all exception thrown by servicecomb using json format and data is CommonExceptionData.

File: integration-tests/it-consumer/src/main/java/org/apache/servicecomb/it/testcase/TestRestServerConfigEdge.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.Test;
+
+public class TestRestServerConfigEdge {
+  static GateRestTemplate rt = (GateRestTemplate) GateRestTemplate.createEdgeRestTemplate("restServerConfig");
+
+  @Test
+  public void testIllegalPathParam() throws IOException {
+    String paramString = "%%A";
+    String requestUri =
+        rt.getUrlPrefix("it-edge", "it-producer", "restServerConfig")
```
Comment (wujimin) [inline integration-tests/it-consumer/src/main/java/org/apache/servicecomb/it/testcase/TestRestServerConfigEdge.java]: we can save the prefix in rt?

File: integration-tests/it-producer/src/main/java/org/apache/servicecomb/it/schema/RestServerConfigSchema.java [hunk 3]
```diff
@@ -31,4 +32,10 @@
   public String testMaxInitialLineLength(@QueryParam("q") String q) {
     return "OK";
   }
+
+  @Path("/testIntPathParam/{intPathParam}")
```
Comment (wujimin) [inline integration-tests/it-producer/src/main/java/org/apache/servicecomb/it/schema/RestServerConfigSchema.java]: intPath?
is it possible just use datatypeSchema's intPath?

File: transports/transport-rest/transport-rest-vertx/src/main/java/org/apache/servicecomb/transport/rest/vertx/RestServerVerticle.java (hunk 1 above)
Comment (yhs0092) [reply transports/transport-rest/transport-rest-vertx/src/main/java/org/apache/servicecomb/transport/rest/vertx/RestServerVerticle.java]: fixed
now response body is `{"message":"unknown error"}`, content-type is `application/json`

File: integration-tests/it-consumer/src/main/java/org/apache/servicecomb/it/testcase/TestRestServerConfigEdge.java (hunk 2 above)
Comment (yhs0092) [reply integration-tests/it-consumer/src/main/java/org/apache/servicecomb/it/testcase/TestRestServerConfigEdge.java]: Done, please review

**Pull request merged**